src/BaseballGame.js

```diff
@@ -0,0 +1,74 @@
+import { MissionUtils } from "@woowacourse/mission-utils";
+import { generateRandomNum } from "./utils/generateRandomNum";
+import { isInputValid, isEndInputValid } from "./utils/validation";
+import { getResult, printResult } from "./utils/getResult";
+
+class BaseballGame {
+  #COMPUTER_NUM_LEN = 3;
+  #ComputerNums;
+  #UserNums;
+
+  setComputerNums(nums) {
+    this.#ComputerNums = nums;
+  }
+
+  setUserNums(nums) {
+    isInputValid(nums, this.#COMPUTER_NUM_LEN);
+    this.#UserNums = nums.split("").map((str) => Number(str));
+  }
+
+  /**
+   * 게임의 전체 진행을 위한 메서드입니다.
+   */
+  async runGame() {
+    await this.startGame();
+    await this.playGame();
+    await this.endGame();
+  }
+
+  /**
+   * 게임 시작을 위한 메서드입니다.
+   * 게임 시작 멘트를 출력하며, 컴퓨터의 숫자 3개를 정합니다.
+   */
+  async startGame() {
+    await MissionUtils.Console.print("숫자 야구 게임을 시작합니다.");
+    this.setComputerNums(await generateRandomNum(1, 9, this.#COMPUTER_NUM_LEN));
+  }
+
+  /**
+   * 게임 플레이를 위한 메서드입니다.
+   * 값을 입력 받고, 받은 값의 유효성 체크와 결과를 출력합니다.
+   */
+  async playGame() {
+    let isUserWin = false;
+    while (!isUserWin) {
+      const USER_INPUT = await MissionUtils.Console.readLineAsync("숫자를 입력해주세요 : ");
+
+      this.setUserNums(USER_INPUT);
+
+      const { strike, ball } = await getResult({
+        computerNums: this.#ComputerNums,
+        userNums: this.#UserNums,
+      });
```

Comment:
```suggestion
      const { strike, ball } = await getResult(this.#ComputerNums, this.#UserNums);
```
인자 넘기실 때 객체로 넘겨주시는 이유가 있으신가요??  
그리고 `getResult` 함수는 비동기적으로 동작하지 않아서 `await`을 사용하지 않아도 될 것 같습니다!

Reply:
혹시 `computerNums`와 `UserNums`의 순서가 바뀌거나 할 수 있어서 이런식으로 작성했습니다.
근데 지금 다시 생각해보니 다른 함수들과는 통일성이 없어서 아쉽네요.. 다른 함수들도 객체로 넘기든가 얘네도 그냥 파라미터로 넘기던가 하는게 좋을 수도 있을 것 같습니다. 이 부분은 리팩토링하면서 좀 더 생각해보겠습니다!
+ 왤케 `await`을 남용한거지ㅠㅠ 더 꼼꼼하게 확인해야겠습니다.. 감사합니다!

Reply:
클래스 문법이여서 그런지 지금 하신 방법도 가독성은 좋은 것 같아요!